core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/BPMBillingProjectDeleteRunner.scala

```diff
@@ -0,0 +1,131 @@
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{
+  WorkspaceManagerResourceJobRunner,
+  WorkspaceManagerResourceMonitorRecord
+}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord.{
+  Complete,
+  Incomplete,
+  JobStatus
+}
+import org.broadinstitute.dsde.rawls.dataaccess.workspacemanager.WorkspaceManagerDAO
+import org.broadinstitute.dsde.rawls.model.{RawlsBillingProjectName, RawlsRequestContext}
+import org.broadinstitute.dsde.rawls.model.CreationStatuses.DeletionFailed
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Delete a BPM billing project after the job corresponding with jobControlId has finished in WSM
+  * (which will be a landing zone deletion job)
+  * Handles both AzureBillingProjectDelete and OtherBpmBillingProjectDelete
+  *   If it's AzureBillingProjectDelete, it means we need to wait for the associated lz job to complete
+  *   If it's OtherBpmBillingProjectDelete, we can delete right away
+  */
+class BPMBillingProjectDeleteRunner(
+  val samDAO: SamDAO,
+  val gcsDAO: GoogleServicesDAO,
+  workspaceManagerDAO: WorkspaceManagerDAO,
+  billingRepository: BillingRepository,
+  billingLifecycle: BillingProjectLifecycle
+) extends WorkspaceManagerResourceJobRunner
+    with LazyLogging
+    with UserCtxCreator {
+
+  override def apply(
+    job: WorkspaceManagerResourceMonitorRecord
+  )(implicit executionContext: ExecutionContext): Future[JobStatus] = {
+    val projectName = job.billingProjectId match {
+      case Some(name) => RawlsBillingProjectName(name)
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneDeleteResult created with id ${job.jobControlId} but no billing project set"
+        )
+        return Future.successful(Complete) // nothing more this runner can do with it
+    }
+    val userEmail = job.userEmail match {
+      case Some(email) => email
+      case None =>
+        logger.error(
+          s"Job to monitor ${job.jobType} $projectName created with id ${job.jobControlId} but no user email set"
+        )
+        val errorMsg =
+          s"Unable to update ${projectName.value} with landing zone deletion status because no user email is stored on monitoring job"
+        return billingRepository
+          .updateCreationStatus(projectName, DeletionFailed, Some(errorMsg))
+          .map(_ => Complete)
+    }
+    getUserCtx(userEmail).transformWith {
+      case Success(userCtx) =>
+        job.jobType match {
+          case AzureBillingProjectDelete =>
+            billingRepository.getLandingZoneId(projectName).flatMap {
+              case Some(landingZoneId) =>
+                handleLandingZoneDeletion(job.jobControlId, projectName, UUID.fromString(landingZoneId), userCtx)
+              case None =>
```

Comment:
I think the intent was to preserve some existing functionality, but it might not have mapped cleanly when splitting into phases.
`OtherBpmBillingProjectDelete` is keyed on the presence of the landing zone, so as written, we _shouldn't_ get into this particular state, and it's really just a loophole in how things have been defined.

I think that the correct way to do this is log the message here (and not on the init side) and proceed to the deletion. Then instead of both job types just have `BpmBillingProjectDelete`.